### DISCUSSION TITLE
#1 First Sprint for AMOGUS starts ✨

### DISCUSSION THREAD

**Discussion by jas20202** [CUMGroup/AMOGUS#3] [Blog]
### But lets start with a little introduction
We are happy to announce that CUMGroup has a new member to it's team. Welcome @WachtelHD 😁!
Our team now consists of:
@AlexMi-Ha (_Architecture; Developer_) 
@Annalytic-programming (_Scruwum-Master; Develowoper_) 
@elnaz-gharoon (_IT-/Datasecurity; Developer_)
@jas20202 (_UX/UI-Design; Developer_)
@WachtelHD (_QA-Engineering; Developer_)

### Vision of our project
> The name AMOGUS is short for **A**dvanced **M**athematic **O**perations & **G**eometrics **U**nder **S**tress
> Like the name already _clearly_ states, we are developing a learning Web Application like Duolingo, but to teach mathematics concepts in a visual fashion.

### Techniques / Platforms we plan to use
> We plan to use the C# programming language with .NET6 and the ASP.NET Framework to develop a modern MVC Application with an interesting and intuitive Graphical User Interface written with the help of Razor Pages templating.

### Scrum setup 
We will be using Jira as our task tracking tool.
Each sprint will be one week long.
And we will have 1 h weekly sprint meetings. 
If needed we will also do daily's.

### Our backlog for this sprint will be
- Analysis on the implementation of the Expression Interpreter
- An(n)alysis how the game should play, what the content is and how the logic functions
- Define a color palette for the application.
- Create Mockups for the Views to visualize the intended design and to make the implementation easier.

Most of the tasks are already in progress and we are confident that we can finish in time 😊
If we manage to finish earlier than expected we will start with the database design and the architecture.

### Our Kanban
![image](https://user-images.githubusercontent.com/99250573/195406187-eaaaafc7-fc5d-4c95-8cc9-104599bfd269.png)

Yours truly,
CUMGroup

**Comment by nuo-li** [comment]
Analysis on the implementation of the Expression Interpreter --> Very good! To what extend, will you analyze?

**Reply by AlexMi-Ha** [reply]
Hi Nuo,
I will be looking into different ways to compare, differentiate, simplify and evaluate two expression trees and their runtimes.
At the time I'm trying to implement my own ideas on how to do it, but I already looked for library alternatives that could handle the work for us!

**Comment by Richone01** [comment]
Your project sounds very interesting. Which topics will it deal with? Like mathematics and geometry is quite a broad range, which will need a lot of different types of exercises. Will all answers be text-based or do you plan to have a special design for every exercise?
 
-DHBWorkout

**Reply by Annalytic-programming** [reply]
Hi Richone01,
we are planning to have a broad exercise spectrum focusing on mainly three categories (mental arithmetic, analysis, algebra). If you are interested in more details on that you can find a basic exercise plan in our repository. 
We are planning to have a few easy exercises generated while run time while we store the complex ones in a database.
We decided to have two ways of answering questions:

- text-based (for this we need the interpreter which was discussed in the comment above yours)
- multiple choice

I hope I could help you with this.
Yours truly,
Anna
CUMGroup

**Comment by jan-nie** [comment]
Dear AMOGUS team,

your vision sounds really interesting. We are looking forward to see your Software Requirements Specification (SRS) to learn more about your C# project. Maybe your software will help us with future math exams ;)

Sol lucet omnibus.

Sincerly, your Framgång (IM SE)
